pkg/eventcollector/eventcollector.go

```diff
@@ -213,41 +237,60 @@ func (c *EventCollector) StartChan() chan bool {
 
 // StartCollection contains logic to run retriever and listener.
 func (c *EventCollector) StartCollection() error {
-	err := c.runRetriever()
-	if err != nil {
-		c.sendStartSignal()
-		return err
-	}
-
-	// nil gotcha appears
-	// https://divan.github.io/posts/avoid_gotchas/#interfaces
-	if c.wsClient == nil || reflect.ValueOf(c.wsClient).IsNil() {
-		log.Infof("Listener is disabled, not starting")
-		c.sendStartSignal()
-		return nil
-	}
-
-	if c.crawlerPubSub != nil {
-		err = c.crawlerPubSub.PublishProcessorTriggerMessage()
+	var once sync.Once
+	for {
+		err := c.runRetriever()
 		if err != nil {
+			once.Do(func() { c.sendStartSignal() })
 			return err
 		}
-	}
 
-	err = c.startListener()
-	if err != nil {
-		c.sendStartSignal()
-		return fmt.Errorf("Error starting listener: err: %v", err)
-	}
+		if c.crawlerPubSub != nil {
+			err = c.crawlerPubSub.PublishProcessorTriggerMessage()
+			if err != nil {
+				log.Errorf("Error publishing trigger message: err: %v", err)
+			}
+		}
 
-	c.sendStartSignal()
+		if c.pollingEnabled {
+			once.Do(func() { c.sendStartSignal() })
+			log.Infof("Polling enabled, waiting...")
+			select {
+			case <-time.After(time.Duration(c.pollingIntSecs()) * time.Second):
```

Comment:
Trying to understand the code -- so where is the part that runs the filterer after each `c.pollingintsecs` duration?

Reply:
yeah, i'll try and make this clearer here.